File: scripts/split_town_json_to_per_wilaya_json.dart
import 'dart:convert';
import 'dart:io';

import 'package:flutter/cupertino.dart';

void main() async {
  Future<String> loadJsonFileString(String path) async {
    return File(path).readAsString();
  }

  final rootDirectory = "C:/Projects/hader-pharm-mobile";
  final jsonString =
      await loadJsonFileString("$rootDirectory/assets/json/town_en.json");

  final List<dynamic> rawList = json.decode(jsonString)['data'];
  final List<Map<String, dynamic>> towns =
      rawList.cast<Map>().map((e) => Map<String, dynamic>.from(e)).toList();

  final Map<int, List<Map<String, dynamic>>> grouped = {};

  for (var town in towns) {
    int wilayaId = town['wilayaId'];
    grouped.putIfAbsent(wilayaId, () => []).add(town);
  }

  final outputDir = Directory('$rootDirectory/assets/json/town/');
  if (!outputDir.existsSync()) {
    outputDir.createSync(recursive: true);
  }

  for (var entry in grouped.entries) {
    final wilayaId = entry.key;
    final fileName = 'town_${wilayaId}_en.json';
    final file = File('${outputDir.path}/$fileName');

    final outputJson = {
      "data": entry.value,
    };

    await file.writeAsString(
      const JsonEncoder.withIndent('  ').convert(outputJson),
    );

    debugPrint('✅ Created: ${file.path}');
  }
}

File: scripts/split_town_json_to_per_wilaya_json.py
import json
import os
from pathlib import Path
import argparse

def group_towns_by_wilaya(input_path: Path, output_dir: Path):
    with open(input_path, 'r', encoding='utf-8') as f:
        json_data = json.load(f)

    raw_list = json_data['data']
    towns = [dict(town) for town in raw_list]

    grouped = {}
    for town in towns:
        wilaya_id = town['wilayaId']
        grouped.setdefault(wilaya_id, []).append(town)

    output_dir.mkdir(parents=True, exist_ok=True)

    for wilaya_id, town_list in grouped.items():
        file_name = f"town_{wilaya_id}_en.json"
        output_path = output_dir / file_name

        with open(output_path, 'w', encoding='utf-8') as f:
            json.dump({"data": town_list}, f, indent=2, ensure_ascii=False)

        print(f"✅ Created: {output_path}")

if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("input_path", help="Path to the source JSON file")
    parser.add_argument("output_dir", help="Directory to output split JSON files")

    args = parser.parse_args()

    group_towns_by_wilaya(Path(args.input_path), Path(args.output_dir))

## python split_town_json_to_per_wilaya_json.py C:/Projects/hader-pharm-mobile/assets/json/town_en.json C:/Projects/hader-pharm-mobile/assets/json/town
